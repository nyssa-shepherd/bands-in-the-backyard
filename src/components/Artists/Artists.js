import React from 'react';

const Artists = () => {
    return (
        <div>
            <h1>Favoritos beyotcho</h1>
        </div>
    )
}

export default Artists;