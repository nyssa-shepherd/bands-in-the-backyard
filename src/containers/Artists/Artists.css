h4 {
  color: red;
  margin: 20px auto;
  width: 350px;
}