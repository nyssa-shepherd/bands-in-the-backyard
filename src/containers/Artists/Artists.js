import React, { Component } from 'react';
import { connect } from 'react-redux';
import CardContainer from '../../containers/CardContainer/CardContainer.js';
import { setArtistInLocation } from '../../actions/actions.js';
import './Artists.css';

export class Artists extends Component {
  render () {  
    return (
      <div className='artist-page'>
        <CardContainer />
      </div>
    );
  }
}

export const mapStateToProps = store => ({
  location: store.location,
  allArtistEvents: store.allArtistEvents
});

export const mapDispatchToProps = dispatch => ({
  setArtistInLocation: artist => dispatch(setArtistInLocation(artist))
});

export default connect(mapStateToProps, mapDispatchToProps)(Artists);