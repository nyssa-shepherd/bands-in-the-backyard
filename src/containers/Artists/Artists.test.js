/* eslint-disable */
import React from 'react';
import { shallow } from 'enzyme';
import { Artists, mapStateToProps, mapDispatchToProps } from './Artists.js';

describe('Artists', () => {
  let mockLocation;
  let mockAllArtistEvents;

  beforeEach(() => {
    mockLocation = 'Denver, CO';
    mockAllArtistEvents = [];
  });

  it('snapshot test', () => {
    const wrapper = shallow(<Artists />);
    expect(wrapper).toMatchSnapshot();
  });

  it('default state', () => {
    const wrapper = shallow(<Artists />);
    expect(wrapper.state().noArtistMessage).toEqual('');
  });

  it('update state', () => {
    const wrapper = shallow(<Artists location={mockLocation} allArtistEvents={mockAllArtistEvents}/>);

    wrapper.instance().setFavoriteArtists();
    expect(wrapper.state().noArtistMessage).toEqual('');
  });

  it('map the store correctly', () => {
    const artist = {
      key: 9,
      name: 'name',
      city: 'city',
      state: 'state',
      date: 'date',
      time: 'time',
      venue: 'venue',
      ticketUrl: 'url'
    };
    const mockStore = {location: 'denver, co', allArtistEvents: [artist]};
    const mapped = mapStateToProps(mockStore);

    expect(mapped.location).toEqual(mockStore.location);
    expect(mapped.allArtistEvents).toEqual(mockStore.allArtistEvents);
  });

  it('call dispatch function after MDTP of setArtistInLocation', () => {
    const mockDispatch = jest.fn();
    const mapped = mapDispatchToProps(mockDispatch);

    mapped.setArtistInLocation();
    expect(mockDispatch).toHaveBeenCalled();
  });
});