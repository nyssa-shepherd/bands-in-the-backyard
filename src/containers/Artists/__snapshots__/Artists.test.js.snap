// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Artists snapshot test 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Artists />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <withRouter(Connect(Search))
          setFavoriteArtists={[Function]}
/>,
        <withRouter(Connect(CardContainer)) />,
      ],
      "className": "artist-page",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "setFavoriteArtists": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <withRouter(Connect(Search))
            setFavoriteArtists={[Function]}
/>,
          <withRouter(Connect(CardContainer)) />,
        ],
        "className": "artist-page",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "setFavoriteArtists": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
